src/apply_state_change.js

```diff
@@ -1,14 +1,47 @@
+import update from 'immutability-helper';
```

Comment:
Immutability helper has a single `update` function that takes a directive. Fortunately the functionality is similar to `immutable-js`. I had to implement "remove" from scratch though.